Iss967: changes to Eventfile needed for overlay producer

bryngemark opened the pull request:
Let's keep the discussion in [ldmx-sw issue 969](https://github.com/LDMX-Software/ldmx-sw/pull/969)

bryngemark committed:
Introduce overlay file specific EventFile

This allows the event number to be reset to the beginning of the
tree if the end is reached before the overlay processor has
finished looping over all the sim tree events.

bryngemark committed:
Comment printout that I would ideally use logger for

bryngemark committed:
Add event skipping method

tomeichlersmith commented on src/Framework/EventFile.cxx:
```diff
@@ -194,7 +198,19 @@ bool EventFile::nextEvent(bool storeCurrentEvent) {
+		if ( isOverlayFile_ ) {
+		  // reset the event counter: reuse events from start of pileup tree
+
+		  /* Here I would have liked to inform the user that a reset happens
+			 but it seems like only Exceptions are foreseen from this class? 
+			 ldmx_log(info)
+		  std::cout  << "Reached end of pileup tree at entry " <<
+			ientry_ << "; resetting event counter!" << std::endl;
+		  */
```
The reason the message passed to `ldmx_log(info)` here is "lossed" is because it is a macro that looks for a variable called `theLog_` to write the message to. You could enable logging for the EventFile class using the enableLogging macro, but I also don't think this is necessary.

bryngemark replied:
Alright. Deleted the comment

tomeichlersmith commented on src/Framework/EventFile.cxx:
```diff
@@ -231,6 +247,12 @@ void EventFile::setupEvent(Event *evt) {
 }
 
+int EventFile::skipToEvent(int offset) {
+  ientry_ = offset % entries_;    // make sure the event number exists
```
Should there be a `-1` here? So that `nextEvent()` moves us to the `offset` event?

bryngemark replied:
Yeah i thought about it but i thought it was a minor detail. You pick a number at random, and then you get that number + 1 in the end, it doesn't matter much. In the (unlikely but occurring) event that you picked the last event number, the tree will wrap. So nothing changes in practice. The shift is arbitrary but deterministic.

bryngemark replied:
Implemented this since I was anyway updating the file.

tomeichlersmith reviewed (commented):
Looks good. The only general comment I have is renaming `isOverlayFile` to something more general like `isLoopable`. This is more descriptive of what that parameter controls and highlights that it could be used for something besides Overlay.

bryngemark committed:
Update overlay file identifier to more general name, review response

Also added a -1 in the event skipping method to ensure that the
offset passed is the event number we end up getting.

bryngemark commented:
I think it's all addressed now.
